forward_command_controller/src/forward_command_controller.cpp

```diff
@@ -130,17 +150,18 @@ controller_interface::return_type ForwardCommandController::update()
     return controller_interface::return_type::SUCCESS;
   }
 
-  const auto joint_num = (*joint_commands)->data.size();
-  if (joint_num != joint_cmd_handles_.size()) {
+  if ((*joint_commands)->data.size() != command_interfaces_.size()) {
     RCLCPP_ERROR_STREAM_THROTTLE(
-      rclcpp::get_logger(
-        logger_name_),
-      *lifecycle_node_->get_clock(), 1000, "command size does not match number of joints");
+      get_lifecycle_node()->get_logger(),
+      *lifecycle_node_->get_clock(), 1000,
+      "command size (" << (*joint_commands)->data.size() <<
+        ") does not match number of interfaces (" <<
+        command_interfaces_.size() << ")");
     return controller_interface::return_type::ERROR;
   }
 
-  for (auto index = 0ul; index < joint_num; ++index) {
-    joint_cmd_handles_[index].set_value((*joint_commands)->data[index]);
+  for (auto index = 0ul; index < command_interfaces_.size(); ++index) {
+    command_interfaces_[index].set_value((*joint_commands)->data[index]);
```

Comment:
@bmagyar This assumes that command_interfaces_ comes ordered as you requested it, but I did not find any documentation explicitly mentioning this, that's why I went with the `get_ordered_interfaces` thing.

we could just explicitely mention that the itnerfaces are provided in the order they were requested, that would simplify things.

Reply:
> we could just explicitely mention that the itnerfaces are provided in the order they were requested, that would simplify things.

I agree this has to be done by CM

Reply:
thanks, resolving this